### PR TITLE
Fix hanging issue under UWP.

### PR DESCRIPTION
**Pull request by xgstation**
Reference: https://github.com/chkr1011/MQTTnet/issues/897
Introduced local variables for IAsyncAction for StreamSocket.ConnectAsync in MqttTcpChannel.Uwp.cs
Using 1st solution as it retains async/await.

### CODE REVIEW

**Comment by JanEggers** [comment]
the proposed fix looks like a workaround can you please try .ConfigureAwait(false) instead.

**Comment by xgstation** [comment]
Yes that also worked out.

**Comment by SeppPenner** [comment]
Looks better now. @JanEggers @chkr1011 What do you think?

File: Source/MQTTnet/Implementations/MqttTcpChannel.Uwp.cs [hunk 1]
```diff
@@ -85,7 +85,7 @@ public async Task ConnectAsync(CancellationToken cancellationToken)
                     socketProtectionLevel = SocketProtectionLevel.Tls10;
                 }
 
-                await _socket.ConnectAsync(new HostName(_options.Server), _options.GetPort().ToString(), socketProtectionLevel);
```
Comment (JanEggers) [inline Source/MQTTnet/Implementations/MqttTcpChannel.Uwp.cs:88]: why did you remove the socketProtectionLevel

Comment (xgstation) [reply Source/MQTTnet/Implementations/MqttTcpChannel.Uwp.cs:88]: Sorry that wasn't my intent, just fixed it.
